crates/ironrdp-web/src/audio.rs

```diff
@@ -0,0 +1,773 @@
+}
+
+#[derive(Debug, Clone)]
+struct BrowserAudioCapabilities {
+    supported_sample_rates: Vec<u32>,
+    #[expect(dead_code)] // Reserved for future bandwidth optimization
+    max_sample_rate: f32,
+    #[expect(dead_code)] // Reserved for future quality fallback
+    min_sample_rate: f32,
+}
+
+impl WebAudioBackend {
+    /// Create a new WebAudioBackend with the specified sample rate
+    pub(crate) fn new(sample_rate: Option<f32>) -> Result<Self, IronError> {
+        let audio_context = AudioContext::new().map_err(|e| {
+            anyhow::Error::msg(format!(
+                "failed to create Web Audio API context (check browser support and user gesture requirement): {e:?}"
+            ))
+        })?;
+
+        let gain_node = audio_context
+            .create_gain()
+            .map_err(|e| anyhow::Error::msg(format!("failed to create Web Audio gain node: {e:?}")))?;
+
+        // Connect gain node to destination
+        gain_node
+            .connect_with_audio_node(&audio_context.destination())
+            .map_err(|e| anyhow::Error::msg(format!("failed to connect gain node to audio destination: {e:?}")))?;
+
+        let context_sample_rate = audio_context.sample_rate();
+        let requested_rate = sample_rate.unwrap_or(context_sample_rate);
+
+        let browser_capabilities = Self::detect_browser_capabilities(&audio_context);
+        let supported_formats = Self::create_supported_formats(requested_rate, &browser_capabilities);
+
+        let audio_queue = AudioQueue {
+            context: audio_context.clone(),
+            current_time: audio_context.current_time(),
+        };
+
+        let context_ready = Rc::new(RefCell::new(false));
+
+        // Set up user gesture activation
+        let backend = Self {
+            audio_context: audio_context.clone(),
+            gain_node,
+            context_sample_rate,
+            supported_formats,
+            browser_capabilities,
+            volume: 1.0,
+            pitch: 1.0,
+            is_active: true, // Mark as active by default
+            context_ready: Rc::clone(&context_ready),
+            pending_audio_data: VecDeque::new(),
+            audio_queue,
+        };
+
+        info!(
+            "WebAudioBackend initialized: {} supported formats, context sample rate: {}Hz",
+            backend.supported_formats.len(),
+            backend.context_sample_rate
+        );
+        for (i, format) in backend.supported_formats.iter().enumerate() {
+            info!(
+                "Audio format {}: {:?} {}Hz {}ch ({}bps)",
+                i, format.format, format.n_samples_per_sec, format.n_channels, format.bits_per_sample
+            );
+        }
+
+        // Set up one-time user gesture listener on document
+        Self::setup_user_gesture_listener(audio_context, context_ready);
+
+        Ok(backend)
+    }
+
+    /// Detect browser audio capabilities
+    /// Simplified to focus on sample rate support since we only use PCM
```

Comment:
suggestion: Proper docstring should be more like this:

```suggestion
    /// Detect browser audio capabilities
    ///
    /// Simplified to focus on sample rate support since we only use PCM.
```

The first paragraph is used as the summary in the overview, and since it’s markdown, the lines a merged together, giving something like:

> Detect browser audio capabilities Simplified to focus on sample rate support since we only use PCM

Whereas my suggestion will result into:

> Detect browser audio capabilities
>
> Simplified to focus on sample rate support since we only use PCM.